### PR TITLE
feat: Support of additional helm values for kubernetes-addons

### LINKED ISSUE #512: [FEATURE] Additional helm values for kubernetes-addons which won't delete the default
#### Is your feature request related to a problem? Please describe
I can't just add additional values for helm. I have to replace the default valyes.yaml with full copy of all values.

Ie.

```terraform
module "aws_eks_addons" {
  external_dns_helm_config = {
    values = [yamlencode({
      sources : ["service", "istio-gateway"]
    })]
  }
}
```

will remove settings from the default values.yaml like provider, zoneIdFilters and aws.region.

#### Describe the solution you'd like

I would like to see such pattern (example for externa-dns):

```terraform
variable "helm_values" {
  type        = any
  default     = {}
  description = "Additional Helm values"
}

locals {
  default_helm_config = {
    values      = [local.default_helm_values, var.helm_values]
  }
  default_helm_values = templatefile("${path.module}/values.yaml", {
    aws_region      = var.addon_context.aws_region_name
    zone_filter_ids = local.zone_filter_ids
  })
}
```

Then:

```terraform
module "aws_eks_addons" {
  external_dns_helm_values = yamlencode({
    sources : ["service", "istio-gateway"]
  )}
}
```

The pattern uses ability of Helm which merges 2 or more values.yaml files.

### PR DESCRIPTION
**Pull request by florentio**
### What does this PR do?

This PR add the ability to set multiple helm values for kubernetes-addons without deleting the default values.

There is a many way to set the values depending on how the consumer is building the raw yaml content. Function like `yamlencode`, `templatefile`, `file` etc... can be used.

Below is an example with 3 different additional values for Karpenter installations

```hcl
 karpenter_helm_config = {
    name       = "karpenter"
    chart      = "karpenter"
    repository = "https://charts.karpenter.sh"
    version    = "0.6.3"
    namespace  = "karpenter"
    values = [
      templatefile("${path.module}/values.yaml", {
        eks_cluster_id       = var.eks_cluster_id,
        eks_cluster_endpoint = var.eks_cluster_endpoint,
        service_account_name = var.service_account_name,
        operating_system     = "linux"
      }),
      file("karpenter-custom-values.yaml"),
      templatefile("karpenter-additional.yaml", {}),
      yamlencode({
        "nodeSelector" : {
          "kubernetes.io/os" : "linux"
        }
      })
    ]
  }
```

### Motivation

- Resolves #512 and #863

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: modules/kubernetes-addons/agones/locals.tf [hunk 1]
```diff
@@ -19,7 +18,8 @@ locals {
 
   helm_config = merge(
     local.default_helm_config,
-    var.helm_config
+    var.helm_config,
+    { values = distinct(concat(try(var.helm_config["values"], []), local.default_helm_values)) }
```
Comment (FernandoMiguel) [inline modules/kubernetes-addons/agones/locals.tf:22]: can you pls break a line for readability

Comment (florentio) [reply modules/kubernetes-addons/agones/locals.tf]: Yes sure, just push the changes

**Comment by florentio** [comment]
Hello, any chance to review/merge this PR?
This PR is justing extending `values` attribute related to helm configurations.
It is going to be hard to keep it up to date whenever a new addon is added.

If it is merged earlier, new addon can take advantage of it and will include that feature from the baseline PR.

Thanks

**Comment by niallthomson** [comment]
From what I can see this still won't let me use the default values file from an addon but provide additional values files? It looks like using `values` in the addon config completely overrides the addons core values file.

**Comment by florentio** [comment]
Hello.
The values file inside the module is always use. 
This PR created an array :
custom values provided by your config + values.yaml defined in the module 
and passed to the helm-addon modules

**Comment by askulkarni2** [comment]
Thanks @florentio. This will be handled in a separate PR. cc @bryantbiggs

**Pull request closed without merging**